### DISCUSSION TITLE
Changes to an atom from atom family are not trigger updates on read only atom

### DISCUSSION THREAD

**Discussion by Cpt-Falcon** [pmndrs/jotai#1886] [General]
I have some code that looks like this:

```
    const [selectedSystemParameters, setSelectedSystemParameters] = useAtom(
        stateSystemParameters({ guid: selectedSystemGuid }),
    );
```

```
    function updateSystem(system: ISystem) {
        setSelectedSystemParameters(new System(system));
    }
```

Then my read only atom:

```
export const systemCollection = atom((get: Getter) => {
    const systemGuidsList = get(stateSystemGuids);
    return systemGuidsList.map((guid) => ({
        guid: guid,
        system: get(stateSystemParameters({ guid: guid })),
    }));
});
```

But calling updateSystem does not trigger the systemCollection atom to update. Any assistance would be appreciated.

**Comment by dai-shi** [comment]
Please check simplified code of atomFamily: https://github.com/pmndrs/jotai/discussions/1323#discussioncomment-3303035

If that doesn't help, you may want to create a minimal reproduction with https://csb.jotai.org .
